REST client/trainings/src/main/java/training/evaluation/training/rest/TrainingResource.java

```diff
@@ -24,12 +25,24 @@ public Training createTraining(@Valid @RequestBody Training training) {
         return repository.findAll();
     }
 
-
     @RequestMapping(method=RequestMethod.DELETE, value="/training/{id}")
     public String delete(@PathVariable String id) {
         Training training = repository.findById(id).get();
         repository.delete(training);
 
         return "Product deleted";
     }
+
+    @RequestMapping(method=RequestMethod.POST, value="/training/{id}")
+    public Training update(@PathVariable String id, @RequestBody Training training) {
+        Training tr = repository.findById(id).get();
+        if(training.getName() != null)
```

Comment:
With Optional interface we can achieve same result. But now it's not required.

Optional<Training> training= repository.findById(id);
if (training.isPresent()) {
	Training _training = training.get();
	_training.setName(training.getName());
        _training.setDescription(training.getDescription());
        _training.setLevel(training.getLevel());
               }